sherpa-onnx/csrc/online-recognizer.cc

```diff
@@ -18,56 +18,50 @@
 
 namespace sherpa_onnx {
 
-std::string OnlineRecognizerResult::AsJsonString() const {
-  std::ostringstream os;
-  os << "{";
-  os << "\"is_final\":" << (is_final ? "true" : "false") << ", ";
-  os << "\"segment\":" << segment << ", ";
-  os << "\"start_time\":" << std::fixed << std::setprecision(2) << start_time
-     << ", ";
-
-  os << "\"text\""
-     << ": ";
-  os << "\"" << text << "\""
-     << ", ";
-
-  os << "\""
-     << "timestamps"
-     << "\""
-     << ": ";
-  os << "[";
-
+/// Helper for `OnlineRecognizerResult::AsJsonString()`
+template<typename T>
+const std::string& VecToString(const std::vector<T>& vec, int32_t precision = 6) {
+  std::ostringstream oss;
+  oss << std::fixed << std::setprecision(precision);
+  oss << "[ ";
   std::string sep = "";
-  for (auto t : timestamps) {
-    os << sep << std::fixed << std::setprecision(2) << t;
+  for (auto item : vec) {
+    oss << sep << item;
     sep = ", ";
   }
-  os << "], ";
-
-  os << "\""
-     << "tokens"
-     << "\""
-     << ":";
-  os << "[";
-
-  sep = "";
-  auto oldFlags = os.flags();
-  for (const auto &t : tokens) {
-    if (t.size() == 1 && static_cast<uint8_t>(t[0]) > 0x7f) {
-      const uint8_t *p = reinterpret_cast<const uint8_t *>(t.c_str());
-      os << sep << "\""
-         << "<0x" << std::hex << std::uppercase << static_cast<uint32_t>(p[0])
-         << ">"
-         << "\"";
-      os.flags(oldFlags);
-    } else {
-      os << sep << "\"" << t << "\"";
-    }
+  oss << " ]";
+  return oss.str();
+}
+
+/// Helper for `OnlineRecognizerResult::AsJsonString()`
+template<>  // explicit specialization for T = std::string
+const std::string& VecToString<std::string>(const std::vector<std::string>& vec,
```

Comment:
```suggestion
std::string VecToString<std::string>(const std::vector<std::string>& vec,
```

Reply:
Please never return a temporary reference from a function.

Reply:
Aha, thank you. 

My bad. I remember in some context returning a local object as a const reference worked, but here I did it badly.
https://stackoverflow.com/questions/13318257/const-reference-to-temporary-vs-return-value-optimization

Best, K.

Reply:
> I remember in some context returning a local object as a const reference worked

Yes, you are right. But in this case, the temporary reference is from the function return value of a temporary object.
Think twice, there are two indirections.

Suppose `obj` is a local variable in a function. It is totally fine to call
```
return obj
```
to return a const reference.

But it is not valid to invoke
```
return obj.someMethod();
```
since `obj` is destroyed when the function returns.

Reply:
Yes, you are right.

The code is fixed now. I tested the JSON output from python, and it looks fine.

Also now the lm_probs and contexts scores are filled only if LM and ContextGraph are used. 
Otherwise there are empty lists to save bandwidth.

Thank you for finding the bug,
Karel